repo.go

```diff
@@ -56,13 +55,26 @@ func Latest(repo Interface, curl *charm.URL) (int, error) {
 // or bundle reference.
 // Charm store references will use the provided parameters.
 // Local references will use the provided path.
-func InferRepository(ref *charm.Reference, charmStoreParams NewCharmStoreParams, localRepoPath string) (Interface, error) {
-	switch ref.Schema {
-	case "cs":
-		return NewCharmStore(charmStoreParams), nil
-	case "local":
-		return NewLocalRepository(localRepoPath)
+func InferRepository(curlStr string, charmStoreParams NewCharmStoreParams, localRepoPath string) (Interface, error) {
+	// First try and interpret as a charm URL.
+	if ref, err := charm.ParseReference(curlStr); err == nil {
+		switch ref.Schema {
+		case "cs":
```

Comment:
This will trigger in the case that curlStr=="wordpress" even if there's a wordpress directory. I think you need to check the path first, but I'm thinking that we should have a parameter that specifies whether to do that so that a client has some choice as to whether arbitrary local paths are checked.

Perhaps allowLocalPath bool?